packages/playground/ssr-html/server.js

```diff
@@ -44,13 +52,18 @@ async function createServer(
 
   app.use('*', async (req, res) => {
     try {
-      let [url] = req.originalUrl.split('?')
-      if (url.endsWith('/')) url += 'index.html'
+      const url = req.originalUrl
```

Comment:
Updated this code to be a bit similar to [SSR docs example](https://vitejs.dev/guide/ssr.html#setting-up-the-dev-server)

This one is failing with unresolved path error.

Reply:
https://github.com/vitejs/vite/pull/7993#issuecomment-1115732748 this can solve the issue. Is that expected behaviour?

Was wondering what exactly is the solution-
- Update docs to add the solution that @poyoho mentioned
- Or fix the resolve logic and automatically handle index.html paths?

Reply:
We can update the docs in v3 if we think that is the correct way forward, but this was working before and documented so we should merge a fix in 2.9 that works as it is currently documented:
```
const url = req.originalUrl
```

Reply:
You can watch `resolve` plugin to know how vite internal resolveId work. So you can set the package.json>main field to avoid the error.

Reply:
@poyoho do you see an issue with adding this logic:
```js
 let [url] = req.originalUrl.split('?') 
 if (url.endsWith('/')) url += 'index.html' 
```
to `transformIndexHtml`?

Or would it later fail in `render` (https://vitejs.dev/guide/ssr.html#setting-up-the-dev-server)
```js
    // 4. render the app HTML. This assumes entry-server.js's exported `render`
    //    function calls appropriate framework SSR APIs,
    //    e.g. ReactDOMServer.renderToString()
    const appHtml = await render(url)
```

Since we are asking users to add the `index.html` by hand, why not add it ourselves when needed so we avoid breaking these cases? Maybe we could emit a warning later saying that in v3 this will not be supported and they have to pass a complete path?